### PR TITLE
Add ament_cmake build_type export in package.xml and install package.xml file

### PR DESCRIPTION
**Pull request by olivier-stasse**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Descrition

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."
This PR is mostly to make the package ```master_board_sdk``` more compliant to ROS-2.
It allows to have a better ROS-2 settings by:
- Add ```ament_cmake```build_type in the file ```package.xml```. Not doing this will prevent ```colcon``` to create the hooks for having the package  in the ROS-2  environment variables.
- Add installation procedure in ```CMakeLists.txt```

## How I Tested

[//]: # "Explain how you tested your changes"
This does not affect the other tests.
Checked if ```package.xml``` was installed, and did:
```
colcon build --packages-select master_board_sdk --cmake-args ' -DCMAKE_BUILD_TYPE=RELEASE'
export COLCON_STRACE=1
source ./install/setup.zsh
```
```master_board_sdk``` appears in ```AMENT_CMAKE_PATH```

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [X] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings). Used pre-commit
- [X] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Review by nim65s — CHANGES_REQUESTED** [review]
This strategy requires ROS 2 tools to be available in order to get an installed package which is compliant with ROS 2.

In eigenpy/hpp-fcl/pinocchio, this is not a requirement.

Could you elaborate on this design choice, and provide the limitations you faced while tring to implement the same logic as in eigenpy/hpp-fcl/pinocchio ?

File: sdk/master_board_sdk/CMakeLists.txt [hunk 1]
```diff
@@ -130,3 +141,8 @@ if(BUILD_TESTING)
   add_unit_test(test_protocol tests/test_protocol.cpp)
   target_link_libraries(test_protocol PRIVATE Catch2::Catch2WithMain)
 endif()
+
+# --- Call ROS-2 if possible ---
+if(ROS_VERSION EQUAL 2)
+  ament_package()
```
Comment (MaximilienNaveau) [inline sdk/master_board_sdk/CMakeLists.txt:147]: if this overwrites the master-board-sdkConfig.cmake we could have issues with the downstream dependencies as jrl-cmakemodules properly writes the dependecies in the Config.cmake.

But for ament to do so you need to call another ament function, I believ it's `ament_export_dependencies(blabla)`.

Maybe you already checked that.

**Comment by olivier-stasse** [comment]
> This strategy requires ROS 2 tools to be available in order to get an installed package which is compliant with ROS 2.
> 
> In eigenpy/hpp-fcl/pinocchio, this is not a requirement.
> 
> Could you elaborate on this design choice, and document the limitations you faced while tring to implement the same logic as in eigenpy/hpp-fcl/pinocchio ?
My initial evaluation was to build master_board_sdk and odri_control_interface in a colcon workspace.
I did this PR based on tests done without updating the CMake version.
Retesting the eigenpy/hpp-fcl/pinocchio approach with a proper CMake version, the use case I was testing is working smoothly. 
Thus I will make the changes accordingly.
But the lines added are the same all over the repositories. It might be worth to include them in jrl-cmakemodules.

File: sdk/master_board_sdk/CMakeLists.txt [hunk 2]
```diff
@@ -130,3 +134,28 @@ if(BUILD_TESTING)
   add_unit_test(test_protocol tests/test_protocol.cpp)
   target_link_libraries(test_protocol PRIVATE Catch2::Catch2WithMain)
 endif()
+
+# --- Call ROS-2 if possible ---
```
Comment (MaximilienNaveau) [inline sdk/master_board_sdk/CMakeLists.txt:138]: this is not accurate anymore, yes?

**Review by MaximilienNaveau — APPROVED** [review]
Sounds good to me

File: sdk/master_board_sdk/CMakeLists.txt (hunk 2 above)
Comment (nim65s) [inline sdk/master_board_sdk/CMakeLists.txt:138]:
```suggestion
# --- Call ROS-2 if possible ---
```

**Review by nim65s — APPROVED** [review]

**Comment by olivier-stasse** [comment]
Thanks @nim65s

**Pull request merged**